### PR TITLE
RTL6c

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by mattheworiordan** [comment]
LGTM but will be revisited following @tcard's updates to the spe

**Comment by tcard** [comment]
PTAL

**Comment by tcard** [comment]
(By the way, that's up to date with ably/docs#112.)

File: Spec/RealtimeClientChannel.swift [hunk 1]
```diff
@@ -1110,6 +1110,208 @@ class RealtimeClientChannel: QuickSpec {
...
+                            channel.publish(nil, data: "message") { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                            expect((client.transport as! TestProxyTransport).protocolMessagesSent.filter({ $0.action == .Message })).to(haveCount(1))
+                        }
+                    }
+
+                    // RTL6c2
+                    context("the message should be queued and delivered as soon as the connection state returns to CONNECTED if the connection is") {
```
Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: The tests in this context test `the message should be queued`, but don't test `and delivered as soon as the connection state returns to CONNECTED`.  Should we?

**Comment by mattheworiordan** [comment]
One comment but otherwise great 👍

File: Spec/RealtimeClientChannel.swift (hunk 1 above)
Comment (tcard) [inline Spec/RealtimeClientChannel.swift]: It is tested. All of them are calling a local `publish` function (line 1150) which publishes and waits for the callback. Or do you want to also subscribe to the channel and check if it's received? I don't think that's necessary, as other tests already prove that the publish callback is called upon successful delivery.

Comment (mattheworiordan) [inline Spec/RealtimeClientChannel.swift]: Yes,  you are right, leave as is

**Pull request merged**